packages/react-composites/src/composites/CallComposite/components/MediaGallery.tsx

```diff
@@ -117,3 +129,29 @@ export const useLocalVideoStartTrigger = (isLocalVideoAvailable: boolean, should
     }
   }, [shouldTransition, isButtonStatusSynced, isPreviewCameraOn, isLocalVideoAvailable, mediaGalleryHandlers]);
 };
+
+/**
+ * Hook to fetch new participant avatar and video gallery displayName information if it is present.
+ * @param onFetchAvatarPersonaData
+ */
+const useOnFetchAvatarPersonaData = (
+  remoteParticipants: VideoGalleryRemoteParticipant[],
+  onFetchAvatarPersonaData?: AvatarPersonaDataCallback
+): VideoGalleryRemoteParticipant[] => {
+  // Sort the remote participants so that they are always in order
+  remoteParticipants.sort((a, b) => (a.userId < b.userId ? 1 : -1));
```

Comment:
I'd recommend against reordering the `remoteParticipants`. This modifies an argument to the function in a surprising way. Side effects like this are hard to understand from the callsite.

I know we talked about sorting the `userIds` passed into `useDataProvider` to better memoize the results, but in this case we use the return value from `useDataProvider` to simply update the incoming `remoteParticipants`. So there is no need to sort the `userIds` here.